src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/validation/AuditValidator.java

```diff
@@ -0,0 +1,123 @@
+
+            // attributes
+            new Tuple<>(Audit.Key.RESOLVE_BULK_REQUESTS, DataType.BOOLEAN),
+            new Tuple<>(Audit.Key.LOG_REQUEST_BODY, DataType.BOOLEAN),
+            new Tuple<>(Audit.Key.RESOLVE_INDICES, DataType.BOOLEAN),
+            new Tuple<>(Audit.Key.EXCLUDE_SENSITIVE_HEADERS, DataType.BOOLEAN),
+
+            // config
+            new Tuple<>(Audit.Key.INTERNAL_CONFIG_ENABLED, DataType.BOOLEAN),
+            new Tuple<>(Audit.Key.EXTERNAL_CONFIG_ENABLED, DataType.BOOLEAN),
+
+            // ignore
+            new Tuple<>(Audit.Key.IGNORE_USERS, DataType.ARRAY),
+            new Tuple<>(Audit.Key.IGNORE_REQUESTS, DataType.ARRAY),
+
+            // compliance read
+            new Tuple<>(Audit.Key.READ_METADATA_ONLY, DataType.BOOLEAN),
+            new Tuple<>(Audit.Key.READ_WATCHED_FIELDS, DataType.ARRAY),
+            new Tuple<>(Audit.Key.READ_IGNORE_USERS, DataType.ARRAY),
+
+            // compliance write
+            new Tuple<>(Audit.Key.WRITE_LOG_DIFFS, DataType.BOOLEAN),
+            new Tuple<>(Audit.Key.WRITE_METADATA_ONLY, DataType.BOOLEAN),
+            new Tuple<>(Audit.Key.WRITE_WATCHED_INDICES, DataType.ARRAY),
+            new Tuple<>(Audit.Key.WRITE_IGNORE_USERS, DataType.ARRAY),
+
+            // immutable
+            new Tuple<>(Audit.Key.IMMUTABLE_INDICES, DataType.ARRAY),
+
+            // salt
+            new Tuple<>(Audit.Key.SALT, DataType.STRING)
+    );
+
+    public AuditValidator(final RestRequest request,
+                          final BytesReference ref,
+                          final Settings esSettings,
+                          final Object... param) {
+        super(request, ref, esSettings, param);
+        KEYS.forEach(x -> allowedKeys.put(x.v1(), x.v2()));
+    }
+
+    @Override
+    public boolean validate() {
+        if (!super.validate()) {
+            return false;
+        }
+
+        if ((request.method() == RestRequest.Method.PUT || request.method() == RestRequest.Method.PATCH)
+                && this.content != null
+                && this.content.length() > 1) {
+            try {
+                final Map<String, Object> contentAsMap = XContentHelper.convertToMap(this.content, false, XContentType.JSON).v2();
+                if (contentAsMap != null) {
+                    // validate the audit categories
+                    if (!validateAuditCategories(contentAsMap)) {
+                        this.errorType = ErrorType.BODY_NOT_PARSEABLE;
+                        return false;
+                    }
+                    // validate salt
+                    if (!validateSalt(contentAsMap)) {
+                        this.errorType = ErrorType.INVALID_SALT;
+                        return false;
+                    }
+                }
+            } catch (NotXContentException e) {
+                //this.content is not valid json/yaml
+                log.error("Invalid xContent: " + e, e);
```

Comment:
Use smart logging. Do you mean to log invalid content or exception? Exception will be logged anyway.

Reply:
Yes.` log.error("Invalid content passed in the request", e);` should be sufficient.